### ISSUE TITLE
redundantParens removes required parentheses for closure parameters with explicitly omitted parameter labels

### ISSUE THREAD

**Issue by ChristopherRogers** [nicklockwood/SwiftFormat#385]
I'm not sure what kind of syntax you call this (it's the first time I've seen it) but the following code will break when the parentheses are removed:

```swift
let closure: (_ success: Bool) -> Void = { (_ success) in }
```

**Comment by nicklockwood** [comment]
@ChristopherRogers thanks fo reporting, I didn't actually realize this was valid syntax (but it sort of makes sense).

I'll fix it, but in the meantime the workarounds I'd suggest are to change the code to either:

```swift
let closure: (_ success: Bool) -> Void = { (_ success: Bool) in }
```

or

```swift
let closure: (_ success: Bool) -> Void = { _ in }
```

Which are both functionally equivalent to what you have now.

**Comment by nicklockwood** [comment]
@ChristopherRogers fixed in 0.40.2.
